src/app/app-routing.module.ts

```
@@ -1,25 +1,36 @@
-import {NgModule} from '@angular/core';
-import {PreloadAllModules, RouterModule, Routes} from '@angular/router';
-import {ErrorComponent} from './scenes/error/unknown-code/error.component';
-import {NotFoundComponent} from "./scenes/error/not-found/not-found.component";
+import { NgModule } from '@angular/core';
+import { PreloadAllModules, RouterModule, Routes } from '@angular/router';
+import { ErrorComponent } from './scenes/error/unknown-code/error.component';
+import { NotFoundComponent } from './scenes/error/not-found/not-found.component';
 
 const routes: Routes = [
     {
         path: '',
         loadChildren: () =>
-            import('./scenes/donation/donation.module').then((m) => m.DonationModule)
-
+            import('./scenes/donation/donation.module').then(
+                (m) => m.DonationModule
+            ),
     },
     {
         path: 'donate',
         loadChildren: () =>
-            import('./scenes/donation/donation.module').then((m) => m.DonationModule)
-
+            import('./scenes/donation/donation.module').then(
+                (m) => m.DonationModule
+            ),
+    },
+    {
+        path: 'registration',
```

we're not really registering a user here... We're merely linking an email to a donation. Interestingly, I'm wondering what will happen when a user comes a second time through this portal with the same email... won't it break ?

---

it will break because then they will have a user for that account. Thats why they have to download the app afterwards to make follow up donations

---

I think we need input from @liesdere2 what the expected flow. With the current API I don't see a better way....

But we might need to add graceful failure if they try again and force them to go into the app....

---

Ideally the change dectector / parser should imho understand the language and ignore non-functional changes (such as whitespace and line ends).

---

For the subsequent donations to the email address: I think there are multiple options

1. silently use an existing user if the email address is known
2. notify the user and let him confirm
3. block the donation

I think option 3 is not desirable. e.g. What if the user has an account and the app installed on his own device, but now is on another device?

---

@gkavanagh What does "break" mean? Does it mean you're not handling the error of the server at all? I would like to see this resolved with option 3, but show a meaningful error message to the user either by first checking whether the email already exists via the API or by capturing the API error correctly.

---

I have put an improvement on the backlog to allow for multiple donations with the same email address.

---

@neeohw interesting to read that you opt for option 3. That would mean that the user is blocked if they have donated before. Or what is it I don't read / understand correctly?

---

No it means that getting the first donation flow to "DONE" is more important than not releasing because a second donation should also be possible.

---

I understand, and that is now already possible. The issue at hand is about subsequent donations. But I think it is a good idea to split that off from the basic story.